rct229/rulesets/ashrae9012019/section11/section11rule13.py

```diff
@@ -0,0 +1,168 @@
+                "Service water-heating energy consumption shall be calculated explicitly based upon the volume of service water heating required and the entering makeup water and the leaving service water-heating temperatures. Entering water temperatures shall be estimated based upon the location. Leaving temperatures shall be based upon the end-use requirements."
+            ),
+            ruleset_section_title="Service Water Heating",
+            standard_section="Table G3.1 #11, baseline column, (e)",
+            is_primary_rule=False,
+            list_path="ruleset_model_descriptions[0]",
+            required_fields={
+                "$": ["calendar"],
+                "calendar": ["is_leap_year"],
+            },
+            data_items={
+                "is_leap_year": (BASELINE_0, "calendar/is_leap_year"),
+            },
+        )
+
+    class RMDRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section11Rule13.RMDRule, self).__init__(
+                rmds_used=produce_ruleset_model_description(
+                    USER=False, BASELINE_0=True, PROPOSED=False
+                ),
+                index_rmd=BASELINE_0,
+                each_rule=Section11Rule13.RMDRule.BuildingRule(),
+                list_path="$.buildings[*]",
+            )
+
+        def create_data(self, context, data):
+            rmd_b = context.BASELINE_0
+            is_leap_year_b = data["is_leap_year"]
+            energy_required_to_heat_swh_use_dict = {}
+            service_water_heating_use_dict = {}
+
+            for building_segment in find_all(
+                "$.buildings[*].building_segments[*]", rmd_b
+            ):
+                service_water_heating_use_list = [
+                    find_exactly_one_service_water_heating_use(rmd_b, swh_use_id)
+                    for swh_use_id in get_swh_uses_associated_with_each_building_segment(
+                        rmd_b, building_segment["id"]
+                    )
+                ]
+
+                service_water_heating_use_dict[
+                    building_segment["id"]
+                ] = service_water_heating_use_list
+                for swh_use in service_water_heating_use_list:
+                    # If no swh use specified or swh use is 0, skip
+                    if swh_use.get("use", 0) == 0:
+                        continue
+                    energy_required_to_heat_swh_use = (
+                        get_energy_required_to_heat_swh_use(
+                            swh_use["id"], rmd_b, building_segment["id"], is_leap_year_b
+                        )
+                    )
+                    if swh_use["id"] not in energy_required_to_heat_swh_use_dict:
+                        energy_required_to_heat_swh_use_dict[swh_use["id"]] = {}
+                    energy_required_to_heat_swh_use_dict[swh_use["id"]][
+                        building_segment["id"]
+                    ] = energy_required_to_heat_swh_use
```

Comment:
I don't quite understand the need for a 3 dimension data structure here.
If we just use 
```python
energy_required_to_heat_swh_use_dict[swh_use["id"]]= energy_required_to_heat_swh_use
```
wouldnt this be enough?

Reply:
This is because [get_energy_required_to_heat_swh_use](https://github.com/pnnl/ruleset-checking-tool/blob/4668dd1ce4435aac886a77539e736d089e791ad6/rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py#L38) takes rmd, swh_use_id, building_segment at the same time as inputs.